src/firebaseFiles/actionInDatabase/addUser.ts

```diff
@@ -0,0 +1,18 @@
+import { doc, setDoc } from 'firebase/firestore'
+import { Socket } from 'socket.io'
+import { DefaultEventsMap } from 'socket.io/dist/typed-events'
+import { FirebaseCollection, Sockets, User } from '../../types'
+import { db } from '../firebaseConfig'
+
+export const addUser = (user: User, socket: Socket<DefaultEventsMap, DefaultEventsMap, DefaultEventsMap, any>) => {
```

Comment:
it should be defined globally to not repeat these types https://socket.io/docs/v4/typescript/